c7n/resources/ecs.py

```diff
@@ -562,6 +584,58 @@ class TaskSubnetFilter(net_filters.SubnetFilter):
     RelatedIdsExpression = "attachments[].details[?name == 'subnetId'].value[]"
 
 
+@Task.filter_registry.register('security-group')
+class TaskSGFilter(net_filters.SecurityGroupFilter):
+
+    ecs_group_cache = None
+
+    RelatedIdsExpression = ""
+    eni_expression = "attachments[].details[?name == 'networkInterfaceId'].value[]"
+    sg_expression = "Groups[].GroupId[]"
+
+    def _get_related_ids(self, resources):
+        groups = dict()
+        group_ids = set()
+        eni_ids = set()
+
+        cexp = jmespath_compile(self.eni_expression)
+        for r in resources:
+            ids = cexp.search(r)
+            if ids:
+                eni_ids.update(ids)
+
+        if eni_ids:
+            client = local_session(self.manager.session_factory).client('ec2')
+            response = client.describe_network_interfaces(
+                NetworkInterfaceIds=list(eni_ids)
+            )
+            if response["NetworkInterfaces"]:
```

Comment:
In scenarios where the `describe_network_interfaces` didn't go through, the `response["NetworkInterfaces"]` expression would error out.